packages/components/src/tooltip/test/index.js

```diff
@@ -23,171 +28,136 @@ describe( 'Tooltip', () => {
 
 			await user.hover( button );
 
-			// Tooltip hasn't appeared yet
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
-
-			act( () => jest.advanceTimersByTime( TOOLTIP_DELAY ) );
-
-			// Tooltip shows after the delay
-			const tooltip = screen.getByText( 'Help text' );
-			expect( tooltip ).toBeVisible();
+			await waitFor( () =>
+				expect( screen.getByText( 'tooltip text' ) ).toBeVisible()
+			);
 
 			// Wait for the tooltip element to be positioned (aligned with the button)
 			await waitFor( () =>
 				expect(
-					getWrappingPopoverElement( tooltip )
+					getWrappingPopoverElement(
+						screen.getByText( 'tooltip text' )
+					)
 				).toBePositionedPopover()
 			);
+
+			await user.unhover( button );
+
+			expect(
+				screen.queryByText( 'tooltip text' )
+			).not.toBeInTheDocument();
 		} );
 
 		it( 'should not show tooltip on focus as result of mouse click', async () => {
-			const user = userEvent.setup( {
-				advanceTimers: jest.advanceTimersByTime,
-			} );
-			const mockOnFocus = jest.fn();
+			const user = userEvent.setup();
 
 			render(
-				<Tooltip text="Help text">
-					<button onFocus={ mockOnFocus }>Hover Me!</button>
+				<Tooltip { ...props }>
+					<Button>Hover Me!</Button>
 				</Tooltip>
 			);
 
-			const button = screen.getByRole( 'button', { text: 'Hover Me!' } );
-			expect( button ).toBeVisible();
-
-			await user.click( button );
-
-			// Tooltip hasn't appeared yet
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
-
-			act( () => jest.advanceTimersByTime( TOOLTIP_DELAY ) );
-
-			// Tooltip still hasn't appeared yet, even though the component was focused
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
-			expect( mockOnFocus ).toHaveBeenCalledWith(
-				expect.objectContaining( {
-					type: 'focus',
-				} )
+			await user.click(
+				screen.getByRole( 'button', { text: 'Hover Me!' } )
 			);
+
+			expect(
+				screen.queryByText( 'tooltip text' )
+			).not.toBeInTheDocument();
 		} );
 
 		it( 'should respect custom delay prop when showing tooltip', async () => {
-			const user = userEvent.setup( {
-				advanceTimers: jest.advanceTimersByTime,
-			} );
-
-			const TEST_DELAY = TOOLTIP_DELAY * 2;
-			const mockOnMouseEnter = jest.fn();
-			const mockOnFocus = jest.fn();
+			const user = userEvent.setup( { delay: TOOLTIP_DELAY } );
+			const CUSTOM_DELAY = TOOLTIP_DELAY + 25;
 
 			render(
-				<Tooltip text="Help text" delay={ TEST_DELAY }>
-					<button
-						onMouseEnter={ mockOnMouseEnter }
-						onFocus={ mockOnFocus }
-					>
-						<span>Hover Me!</span>
-					</button>
+				<Tooltip { ...props } delay={ CUSTOM_DELAY }>
+					<Button>Hover Me!</Button>
 				</Tooltip>
 			);
 
 			const button = screen.getByRole( 'button', { name: 'Hover Me!' } );
-			expect( button ).toBeVisible();
 
 			await user.hover( button );
 
-			// Tooltip hasn't appeared yet
-			expect( mockOnMouseEnter ).toHaveBeenCalledTimes( 1 );
```

Comment:
We're removing some more mock call assertions in this test, could you elaborate on the rationale?

Reply:
I thought these were important to test in `should not show tooltip if the mouse leaves the anchor before the tooltip has shown` but not as much here. So I thought these could be simplified to just what needed to be tested, the custom delay.

I'd love to learn why it would be good to have these here as well, but then not in every test (i.e. the test `should render the tooltip when anchor is hovered`).